kernel/memory.c

```diff
@@ -1,13 +1,208 @@
-#ifndef MEMORY_C
-#define MEMORY_C
+#ifndef LINKED_LIST_C
```

Comment:
Keep this as MEMORY_C, as the guards are based on the relative path from `kernel/`, `kernel/memory.c` thus becomes MEMORY_C